server/mediaserver.go

```diff
@@ -510,10 +511,88 @@ func getRTMPStreamHandler(s *LivepeerServer) func(url *url.URL) (stream.RTMPVide
 
 //End RTMP Handlers
 
+func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
+
+	// we read this unconditionally, mostly for ffmpeg
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		glog.Info(err)
+		return
+	}
+	r.Body.Close()
+
+	if ".ts" != path.Ext(r.URL.Path) {
+		// ffmpeg sends us a m3u8 as well, so ignore
+		// Alternatively, reject m3u8s explicitly and take any other type
+		// TODO also look at use content-type
+		return
+	}
+
+	mid := parseManifestID(r.URL.Path)
+	s.connectionLock.RLock()
+	cxn, exists := s.rtmpConnections[mid]
+	s.connectionLock.RUnlock()
+
+	// Check for presence and register if a fresh cxn
+	if !exists {
+		appData := (createRTMPStreamIDHandler(s))(r.URL)
+		if appData == nil {
+			return
+		}
+		st := stream.NewBasicRTMPVideoStream(appData)
+		cxn, err = s.registerConnection(st)
+		// TODO monitor for activity and clean up
```

Comment:
activity monitor and cleanup should block this PR

Reply:
Yes, it's mentioned under "State Cleanup"